File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        // pair of (matching number, original index)
        unordered_map<int, int> checked;
        for (int i=0; i < nums.size(); i++) {
            if (checked.find(nums[i]) != checked.end()) {
                return {checked[nums[i]], i};
            }
            int match = target - nums[i];
            checked.insert({match, i});
            //cout << checked.find(match) << endl ;
        }
        return {};
    }
};

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        i, j = 0, len(numbers)-1
        p1 = numbers[i]
        p2 = numbers[j]

        while i < len(numbers)-1 and j > 0:
            if p1 + p2 < target:
                i += 1
                p1 = numbers[i]
            elif p1 + p2 > target:
                j -= 1
                p2 = numbers[j]
            else:
                return [i+1, j+1]
        
        return None

